R/dependecies.R

```diff
@@ -0,0 +1,437 @@
+#' Get dependencies for all elements in lock
+get_ordered_dependencies <- function(lock) {
+   cat(crayon::blue("Retrieving dependency info..."))
+   names(lock) <- vapply(lock, `[[`, character(1), "name")
+   get_dependencies_for_list(lock, lock, list())
+}
+
+#' Recursive function to take a list and lock and extract dependencies, sorting
+#' along the way using the combine_dependencies function.
+#' @param master_list list.  Packages to go through and parse dependencies from
+#' @param lock list.  Original list of locked packages
+#' @param previously_parsed_deps.  List of packages and their dependencies that
+#'   we have already parsed out of their respective description files
+get_dependencies_for_list <- function(master_list, lock, previously_parsed_deps) {
```

Comment:
Yep.  That's about right.  Because we use version comparison to decide on the "winners" of deciding between dependencies, the dependency list we wind up with will be stable and unmodified in the last pass.  It will also be much faster since once a package is parsed for it's dependencies they are saved into the previously_parsed_dependencies list.